azure-cli.sh

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env bash
+set -e
+
+echo "Azure container registry (ACR) name ( the X part in X.azurecr.io ):"
+read AZ_REGISTRY
+
+echo "Azure Tenant ID :"
+read AZ_TENANT
+
+echo "Azure username :"
+read AZ_USERNAME
+
+echo "Azure password :"
+read -s AZ_PASSWORD
+
+sudo apt-get update && sudo apt-get install -y curl apt-transport-https lsb-release gnupg
+
+curl -sL https://packages.microsoft.com/keys/microsoft.asc | gpg --dearmor | sudo tee /etc/apt/trusted.gpg.d/microsoft.asc.gpg > /dev/null
+
+echo "deb [arch=amd64] https://packages.microsoft.com/repos/azure-cli/ $(lsb_release -cs) main" | sudo tee /etc/apt/sources.list.d/azure-cli.list
+
+sudo apt-get update && sudo apt-get install azure-cli
+
+# If the azure username is set
+if [ -n "$AZ_USERNAME" ] && [ -n "$AZ_PASSWORD" ] && [ -n "$AZ_TENANT" ] && [ -n "$AZ_REGISTRY" ]; then
+  az login --service-principal --username "$AZ_USERNAME" -p "$AZ_PASSWORD" --tenant "$AZ_TENANT"
+
+  CRONTAB_CONTENT=$(crontab -l 2>/dev/null || true)
+
+  # Test if user crontab contains `az acr login --name $AZ_REGISTRY`. If not, add it.
+  if ! echo "$CRONTAB_CONTENT" | grep -q "az acr login --name $AZ_REGISTRY"; then
```

Comment:
It may return a false positive, because it looks into the crontab if it CONTAINS that string (as specified above).
But you know too well that "contains" may be tricky : In the rare case where I already have a "inetum1" registry and I want to add a "inetum" registry, it will not add it to the crontab.

Don't know if you can specify something like $ for the end of line, or maybe enclose `$AZ_REGISTRY` inside quotes ? like `az acr login --name \"$AZ_REGISTRY\"` (and adapt below).

You might also consider I'm nitpicking and ignore this message :)